File: leetcode/Hot-100/fzecs.py
# -*- coding: utf-8 -*-
""" 操作给定的二叉树，将其变换为源二叉树的镜像。
比如：    源二叉树 
            8
           /  \
          6   10
         / \  / \
        5  7 9 11
        镜像二叉树
            8
           /  \
          10   6
         / \  / \
        11 9 7  5 """

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return
        stack  = [root]
        while stack:
            cur = stack.pop()
            if cur.left: stack.append(cur.left)
            if cur.right: stack.append(cur.right)
            cur.left, cur.right = cur.right, cur.left
        return root

File: leetcode/Hot-100/c++/fzecs.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
 #include<stack>
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr){
            return root;
        }
        stack<TreeNode*> s;
        s.push(root);
        while(!s.empty()){
            int s_size = s.size();
            for(int i=0; i<s_size; i++){
               TreeNode* cur = s.top();
               s.pop();
               if (cur->left) s.push(cur->left);
               if (cur->right) s.push(cur->right);
               TreeNode* temp = cur->left;
               cur->left = cur->right;
               cur->right = temp;
            }
        }
        return root;

    }
};